File: src/main/java/io/tripled/social/client/infrastructure/InMemorySocialNetworkRepository.java
package io.tripled.social.client.infrastructure;

import io.tripled.social.client.domain.DateTimeProvider;
import io.tripled.social.client.domain.SocialNetwork;
import io.tripled.social.client.domain.SocialNetworkRepository;

public class InMemorySocialNetworkRepository implements SocialNetworkRepository {

  private final SocialNetwork socialNetwork;

  public InMemorySocialNetworkRepository(DateTimeProvider dateTimeProvider) {
    socialNetwork = new SocialNetwork(new InMemoryMessageRepository(), dateTimeProvider);
  }

  @Override
  public SocialNetwork get() {
    return socialNetwork;
  }
}

File: src/main/kotlin/io/tripled/social/client/infrastructure/InMemorySocialNetworkRepository.kt
package io.tripled.social.client.infrastructure

import io.tripled.social.client.domain.DateTimeProvider
import io.tripled.social.client.domain.SocialNetwork
import io.tripled.social.client.domain.SocialNetworkRepository

class InMemorySocialNetworkRepository(dateTimeProvider: DateTimeProvider) : SocialNetworkRepository {
    private val socialNetwork: SocialNetwork =
        SocialNetwork(InMemoryMessages(), InMemoryRelationships(), dateTimeProvider)

    override fun get(): SocialNetwork {
        return socialNetwork
    }
}
